### PR TITLE
feature: history component for warranty transactions

### PR DESCRIPTION
**Pull request by DecFox**
## Checklist
- [x] I have read the contribution guidelines
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior
- [x] Underlying issue for this pull request: https://github.com/Mithra-Warranty/Mithra/issues/1

## Description

This introduces a table component exhibiting the token history.

### CODE REVIEW

**Comment by adnanhashmi09** [comment]
The new component is throwing Hydration error.
```bash
Unhandled Runtime Error

Error: Hydration failed because the initial UI does not match what was rendered on the server.

See more info here: https://nextjs.org/docs/messages/react-hydration-error
Call Stack
throwOnHydrationMismatch
node_modules/react-dom/cjs/react-dom.development.js (12507:0)
tryToClaimNextHydratableInstance
node_modules/react-dom/cjs/react-dom.development.js (12535:0)
updateHostComponent
node_modules/react-dom/cjs/react-dom.development.js (19902:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (21618:0)
callCallback
node_modules/react-dom/cjs/react-dom.development.js (4164:0)
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4277:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (27451:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (26557:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (26466:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (26434:0)
performConcurrentWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (25738:0)
workLoop
node_modules/scheduler/cjs/scheduler.development.js (266:0)
flushWork
node_modules/scheduler/cjs/scheduler.development.js (239:0)
performWorkUntilDeadline
node_modules/scheduler/cjs/scheduler.development.js (533:0)
```

This error probably occurs due to the incorrect usage of table tags. The dom tree expected by the browser is not being generated, so it throws a hydration error.

File: components/tables/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    event?: string;
+    to?: string;
+    from?: string;
+    date?: string
+}
+
+function Table({headers, data} : {headers: Header, data: any}) { 
+    return (
+        <table className={styles.table}>
+            <tr className={styles.table__row__header}>
```
Comment (adnanhashmi09) [inline components/tables/Table.tsx]: Wrap `<tr>` with `<thead></thead>` tag.

```js
<thead>
  <tr className={styles.table__row__header}>
      {Object.values(headers).map((header: string) => {
          return (
              <th className={styles.table__header} key={header}>{header}</th>
          );
      })}
  </tr>
</thead>

```

File: components/tables/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                {Object.values(headers).map((header: string) => {
+                    return (
+                        <th className={styles.table__header} key={header}>{header}</th>
+                    );
+                })}
+            </tr>
+            {data.map((entry: any, id: number) => {
```
Comment (adnanhashmi09) [inline components/tables/Table.tsx]:
```js

<tbody>
  {data.map((entry: any, id: number) => {
      return (
          <tr className={styles.table__row__content} key={id}>
              {Object.keys(headers).map((header: string) => 
                  Object.entries(entry).map(([key, val]: [key: string, val: any]) => {
                      if(key == header) {
                          return (
                              <th className={styles.table__content} key={key}>
                                  {key === 'txn' ? <a href="#">{val}</a> : val}
                              </th>
                          )
                      }
                  }
              )
          )}
          </tr>
      );
  })}
</tbody>


```

**Comment by adnanhashmi09** [comment]
Also the heading and the cell data are not aligned. Either center align them all or left align them. Experiment with what looks best.

File: components/tables/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                return (
+                    <tr className={styles.table__row__content} key={id}>
+                        {Object.keys(headers).map((header: string) => 
+                            Object.entries(entry).map(([key, val]: [key: string, val: any]) => {
+                                if(key == header) {
+                                    return (
+                                        <th className={styles.table__content} key={key}>
```
Comment (adnanhashmi09) [inline components/tables/Table.tsx]: instead of `<th>` tags use `<td>` tags. They make more sense here as there are regular cell data and not headings.

**Comment by DecFox** [comment]
Thank you for the feedback! Made the required changes.

**Review by adnanhashmi09 — APPROVED** [review]
Great!

**Pull request merged**